spec/fixtures/mock_upstream.lua

```diff
@@ -1,12 +1,94 @@
-local utils = require "kong.tools.utils"
+local utils      = require "kong.tools.utils"
+local cjson_safe = require "cjson.safe"
+local cjson      = require "cjson"
+local ws_server  = require "resty.websocket.server"
+
+
+local function string_trim(s)
+  local _,i1 = s:find("^%s*")
+  return s:sub(i1 + 1, s:find("%s*$") - 1)
+end
+
+
+local function string_beginswith(s, prefix)
+  return s:sub(1, #prefix) == prefix
+end
```

Comment:
To avoid maintaining our own versions of those small utilities (and avoid the need for testing them, because other contributors will see them, copy paste them, and use them), we introduced Penlight as a dependency a while ago - for cold code paths only though :) Will take care of it as well!